src/FSharpx.Extras/ComputationExpressions/Option.fs

```diff
@@ -132,6 +132,12 @@ module Option =
         | Ok a -> Some a
         | _ -> None
 
+    /// Maps Unchecked object when null to None, otherwise Some value.
+    /// It's useful when getting data from external sources, pe.
+    let inline ofUnchecked (x: 'a when 'a : not struct) =
+        match box x = null with
```

Comment:
I did a little benchmarking (will upload it tomorrow) and it seems that it would be better in terms of performance to change it to:
```fsharp
match box x with
| null -> None
| _ -> Some x
```

Reply:
See https://github.com/gdziadkiewicz/FSharpx.Extras/blob/OfUnchekedBenchmark/benchmark/BenchmarkDotNet.Artifacts/results/benchmark.OfUncheckedBenchmark-report-github.md

and the content of the branch to see how I tried to benchmark it. The results seem to agree with https://stackoverflow.com/a/6818329/2976345 which makes me more confident about them.

Reply:
Seems a good way to find better solution :D